roqoqo/tests/integration/operations/four_qubit_gate_operations.rs

```diff
@@ -0,0 +1,1282 @@
+// Copyright © 2021-2024 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Integration test for public API of four qubit gate operations
+
+#[cfg(feature = "json_schema")]
+use jsonschema::{Draft, Validator};
+use ndarray::array;
+use num_complex::Complex64;
+use qoqo_calculator::{Calculator, CalculatorFloat};
+use roqoqo::operations::*;
+use roqoqo::Circuit;
+#[cfg(feature = "json_schema")]
+use schemars::schema_for;
+use std::collections::{HashMap, HashSet};
+
+#[cfg(feature = "json_schema")]
+use test_case::test_case;
+
+#[test]
+fn test_circuit_triple_controlled_x() {
+    let gate = TripleControlledPauliX::new(0, 1, 2, 3);
+    let c = gate.circuit();
+
+    let mut comparison_circuit = Circuit::new();
+    comparison_circuit += CNOT::new(0, 3);
+    comparison_circuit += CNOT::new(0, 1);
+    comparison_circuit += CNOT::new(1, 3);
+    comparison_circuit += CNOT::new(0, 1);
+    comparison_circuit += CNOT::new(1, 3);
+    comparison_circuit += CNOT::new(1, 2);
+    comparison_circuit += CNOT::new(2, 3);
+    comparison_circuit += CNOT::new(0, 2);
+    comparison_circuit += CNOT::new(2, 3);
+    comparison_circuit += CNOT::new(1, 2);
+    comparison_circuit += CNOT::new(2, 3);
+    comparison_circuit += CNOT::new(0, 2);
+    comparison_circuit += CNOT::new(2, 3);
+
+    assert!(c == comparison_circuit);
+}
+
+#[test]
+fn test_matrix_output_triple_controlled_x() {
```

Comment:
I would suggest moving the matrix tests right to the bottom of the file, for easier legibility